### PR TITLE
Patch update according class diagram:  qik_2s12v10_c.hpp, qik_2s12v10_c.cpp, platform.hpp

### PR DESCRIPTION
**Pull request by ForgeRoot**
in qik_2s12v10_c.hpp
- deleted turn(int8_t degrees) 
- the turn method is deleted and will be transferred to beetle, turtle, rhino classes
- I changed the init() to another place in the same class according to class diagram

In moving platform.hpp
- The parameters are private 
- methods set_speed, set_steering are public 
- method turn  is added

### CODE REVIEW

File: code/headers/qik_2s12v10.hpp [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 /// @file
 
-namespace r2d2::moving_platform {
+namespace r2d2 {
```
Comment (verae98) [inline code/headers/qik_2s12v10.hpp]: it still needs to be r2d2::moving_platform

File: code/headers/qik_2s12v10.hpp [hunk 2]
```diff
@@ -65,10 +70,6 @@ namespace r2d2::moving_platform {
         /// is full power forward.
         void set_m1_speed(const int8_t &new_speed);
 
-        /// @brief
-        /// Initializes the qik2s12v10 by resetting it and then setting the baud
-        /// rate.
-        void init();
```
Comment (verae98) [inline code/headers/qik_2s12v10.hpp:71]: Why is init deleted? We need this function to start the qik. It is in the classdiagram

**Review by verae98 — CHANGES_REQUESTED** [review]

File: code/headers/moving_platform.hpp [hunk 3]
```diff
@@ -50,11 +61,6 @@ namespace r2d2::moving_platform {
         virtual void move(const int8_t &distance) = 0;
         virtual void move(const int8_t &x, const int8_t &y) = 0;
 
-    public:
-        moving_platform_c(base_comm_c &comm) : base_module_c(comm) {
```
Comment (verae98) [inline code/headers/moving_platform.hpp:54]: Why is this deleted? This is necessary for the canbus

**Review by verae98 — CHANGES_REQUESTED** [review]

File: code/headers/qik_2s12v10.hpp (hunk 2 above)
Comment (ForgeRoot) [reply code/headers/qik_2s12v10.hpp:71]: It's not deleted it's changed his place inside the qik_2s12v10.hpp as second public method

File: code/headers/qik_2s12v10.hpp (hunk 1 above)
Comment (ForgeRoot) [reply code/headers/qik_2s12v10.hpp]: this is added in the last commit

File: code/headers/moving_platform.hpp (hunk 3 above)
Comment (ForgeRoot) [reply code/headers/moving_platform.hpp:54]: It's not deleted it's changed his place inside the moving_platform.hpp as first public method

Comment (verae98) [reply code/headers/moving_platform.hpp:54]: Ah ok, my bad

**Review by verae98 — APPROVED** [review]
